### PR TITLE
adding storage MaxConnections workaround

### PR DESCRIPTION
**Pull request by brettsam**

### CODE REVIEW

**Review by cgillum — APPROVED** [review]
This seems good to me. Maybe we could add a trace if the reflection fails so that we can detect if the underlying storage implementation changed?

**Comment by brettsam** [comment]
I've beefed this up with:
- logging
- dynamic-only setting of the value
- swallowing (and logging) errors if it fails -- tests showed this

**Comment by fabiocav** [comment]
@brettsam this has been addressed by your other changes, right? Can we close this now?

File: WebJobs.sln [hunk 1]
```diff
@@ -53,9 +53,15 @@ Project("{6EC3EE1D-3C4E-46DD-8F32-0CC8E7565705}") = "FSharpFunctions", "test\Tes
...
+		src\Microsoft.Azure.WebJobs.Shared\WebJobs.Shared.projitems*{39554abe-d7b7-45ff-8d7c-56432abcde3d}*SharedItemsImports = 5
 		src\Microsoft.Azure.WebJobs.Shared.Storage\Microsoft.Azure.WebJobs.Shared.Storage.projitems*{6bed7f8a-a199-4d9d-85d1-6856ee3292c6}*SharedItemsImports = 13
 		src\Microsoft.Azure.WebJobs.Protocols\Microsoft.Azure.WebJobs.Protocols.projitems*{6fcd0852-6019-4cd5-9b7e-0de021a72bd7}*SharedItemsImports = 13
+		src\Microsoft.Azure.WebJobs.Shared.Storage\Microsoft.Azure.WebJobs.Shared.Storage.projitems*{a9733406-267c-4a53-ab07-d3a834e22153}*SharedItemsImports = 5
+		src\Microsoft.Azure.WebJobs.Shared\WebJobs.Shared.projitems*{a9733406-267c-4a53-ab07-d3a834e22153}*SharedItemsImports = 5
 		src\Microsoft.Azure.WebJobs.Shared\WebJobs.Shared.projitems*{add036f5-2170-4b05-9e0a-c2ed0a08a929}*SharedItemsImports = 13
+		src\Microsoft.Azure.WebJobs.Shared.Storage\Microsoft.Azure.WebJobs.Shared.Storage.projitems*{ded33098-fe99-436c-96cc-b59a30bef027}*SharedItemsImports = 5
+		src\Microsoft.Azure.WebJobs.Shared\WebJobs.Shared.projitems*{ded33098-fe99-436c-96cc-b59a30bef027}*SharedItemsImports = 5
 	EndGlobalSection
```
Comment (brettsam) [inline WebJobs.sln:65]: I don't know what's up with this sln... gonna see if I can undo it.

Comment (brettsam) [reply WebJobs.sln:65]: not sure... just keeps re-adding it when I open VS.

**Review by fabiocav — APPROVED** [review]

**Pull request merged**